index.js

```diff
@@ -61,17 +61,20 @@ const sendRequest = (async (method, params) => {
   })
 })
 
-const decodeTransaction = async (transaction_bytecode) => {
-  return await sendRequest('decoderawtransaction', [transaction_bytecode])
-}
-
 const getTransactionData = async (transaction_hashes) => {
-  const decodedTransactions = []
+  let decodedTransactions = []
+  listPromises = []
   for (const transaction_hash of transaction_hashes){
-    let transactionBytecode = await sendRequest('getrawtransaction', [transaction_hash])
-    let decodedTransaction = await decodeTransaction(transactionBytecode)
-    decodedTransactions.push(decodedTransaction)
+    let promise = sendRequest('getrawtransaction', [transaction_hash]).then(value =>{
+      sendRequest('decoderawtransaction', [value]).then(decodedTransaction =>{
+        listPromises.push(decodedTransaction);
```

Comment:
I think at this point you need to:
`return decodedTransaction`
this would be the result of the promise once evaluated.